tests/JavaTest.java

```diff
@@ -347,6 +350,41 @@ static void TestCreateUninitializedVector() {
         TestEq(ByteBuffer.wrap(inventory), monsterObject.inventoryAsByteBuffer());
     }
 
+    static void TestReflection() {
+        byte[] data = null;
+        File file = new File("monster_test.bfbs");
+        RandomAccessFile f = null;
+        try {
+            f = new RandomAccessFile(file, "r");
+            data = new byte[(int)f.length()];
+            f.readFully(data);
+            f.close();
+        } catch(IOException e) {
+            System.err.println("FlatBuffers test: couldn't read binary schema file");
+            return;
+        } finally {
+            if (f != null) {
+                try {
+                    f.close();
+                } catch (IOException e) {
+                    //ignored
+                }
+            }
+        }
+
+        // Now test it:
+
+        ByteBuffer bb = ByteBuffer.wrap(data);
+        Schema schema = Schema.getRootAsSchema(bb);
+        reflection.Object rootTable = schema.rootTable();
+        TestEq(rootTable.name(), "Monster");
```

Comment:
Hi Wouter,
This is the plan to add the same set of tests than the C++ part. But I'm doing it one step at a time ;). For now we (with Baptiste) are trying to solve the first issue we encounter: being able to generate java code in a prefixed package (Step 1). Then as soon as your are satified with this we will move forward on the testing and implementation part.